src/ecash/config.rs

```diff
@@ -0,0 +1,77 @@
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct LnConfig {
+    pub cln_path: Option<PathBuf>,
+    pub invoice_description: Option<String>,
+    pub fee_percent: f64,
+    pub reserve_fee_min: cdk::Amount,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum LnMessage {
+    PaymentReceived,
+}
+
+// assumes /tmp/plebpool exists
+// todo: unify all paths into one single root path
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct MintConfig {
+    pub url: String,
+    #[serde(default = "path_default")]
+    pub db_path: PathBuf,
+    #[serde(default = "last_pay_path")]
+    pub last_pay_path: String,
+    pub listen_host: String,
+    pub listen_port: u16,
+    pub mnemonic: String,
+    #[serde(default = "derivation_path_default")]
+    pub derivation_path: String,
+    #[serde(default = "max_order_default")]
+    pub max_order: u8,
+    pub min_fee_reserve: cdk::Amount,
+    pub min_fee_percent: f32,
+    pub ln: LnConfig,
+}
+
+fn path_default() -> PathBuf {
+    PathBuf::from_str("/tmp/plebpool/mint.redb").unwrap()
+}
+
+fn derivation_path_default() -> String {
+    "0/0/0/0".to_string()
+}
+
+fn max_order_default() -> u8 {
+    32
+}
+
+fn last_pay_path() -> String {
+    "/tmp/plebpool/last_pay".to_string()
+}
+
```

Comment:
```suggestion
// assumes /tmp/plebpool exists
// todo: unify all paths into one single root path
#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct MintConfig {
    pub url: String,
    pub db_path: PathBuf,
    pub last_pay_path: String,
    pub listen_host: String,
    pub listen_port: u16,
    pub mnemonic: String,
    pub derivation_path: String,
    pub max_order: u8,
    pub min_fee_reserve: cdk::Amount,
    pub min_fee_percent: f32,
    pub ln: LnConfig,
}

impl Default for MintConfig {
    fn default() -> Self {
        db_path: PathBuf::from_str("/tmp/plebpool/mint.redb").unwrap(),
        derivation_path: "0/0/0/0".to_string(),
        max_order: 32,
        last_pay_path: "/tmp/plebpool/last_pay".to_string(),
        .. Default::default()
    }
}
```

Ref: https://doc.rust-lang.org/std/default/trait.Default.html